### PR TITLE
Major updates

### PR DESCRIPTION
**Pull request by nickhsine**
* Fix /photography endpoint
* remove proxy server. call api server directly
* add entitiesForAuthors field in redux state to store entities of
author and author-articles reducers.
* add process.env.NODE_ENV in dev webpack config
* remove /api folders
* Remove unused actions and reducers
* Update .travis.yml.

### CODE REVIEW

**Comment by nickhsine** [comment]
@YuCJ 幫忙看一下這個 PR ，有改到作者頁的部分。

File: src/actions/author-articles.js [hunk 1]
```diff
@@ -54,8 +55,8 @@ export function fetchAuthorCollection({ targetPage, authorId, returnDelay }) {
       page: targetPage
     }
     // Trans searchParas object to url parameters:
-    let urlParasString = urlParasToString(searchParas)
-    let url = formatUrl(`searchPosts?${urlParasString}`)
+    let urlParasString = qs.stringify(searchParas)
+    let url = formAPIURL(`search/posts?${urlParasString}`, false)
```
Comment (YuCJ) [inline src/actions/author-articles.js]: `formAPIURL` only takes one argument?
https://github.com/twreporter/twreporter-redux/blob/master/src/utils/form-api-url.js

File: src/actions/authors.js [hunk 2]
```diff
@@ -50,8 +52,8 @@ export function searchAuthors({ keywords, targetPage, returnDelay }) {
       page: targetPage
     }
     // Trans searchParas object to url parameters:
-    let urlParasString = urlParasToString(searchParas)
-    let url = formatUrl(`searchAuthors?${urlParasString}`)
+    let urlParasString = qs.stringify(searchParas)
+    let url = formAPIURL(`search/authors?${urlParasString}`, false)
```
Comment (YuCJ) [inline src/actions/authors.js]: `formAPIURL` only takes one argument?
https://github.com/twreporter/twreporter-redux/blob/master/src/utils/form-api-url.js

File: src/actions/author-articles.js (hunk 1 above)
Comment (nickhsine) [inline src/actions/author-articles.js]: sorry, forget to commit the latest update.

Comment (YuCJ) [inline src/actions/author-articles.js]: 🙆‍♂️ 👍

**Pull request merged**